### PR TITLE
WT-13860 Fix timestamp check on wrong prepared update

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**
The assumption in the code flow that is wrong is during the prepare update resolution by a transaction, the update chain will either have the aborted updates or only the prepared updates by that transaction.

All the prepared updates from a transaction are resolved as part of the resolution of the first operation on that key and the subsequent operation resolution on the same key will be no-op. Hence even when the commit of a transaction with multiple operations on the same key is in progress the prepared updates of that transaction in the update chain of a particular key could be resolved and hence new prepared updates by other transactions could be added to the update chain.

### CODE REVIEW

**Comment by bvpvamsikrishna** [comment]
[Patch build](https://spruce.mongodb.com/version/675b4ef090d06800072ee7e4/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&statuses=started) with this code change, that was reproducing the issue.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/txn/txn.c [hunk 1]
```diff
@@ -1219,10 +1219,13 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           txn->id, __wt_timestamp_to_string(txn->prepare_timestamp, ts_string[0]));
 
     /*
-     * Aborted updates can exist in the update chain of our transaction. Generally this will occur
-     * due to a reserved update. As such we should skip over these updates.
+     * Aborted updates can exist in the update chain of our transaction due to reserved update. All
+     * the prepared updates on a key by a transaction will be resolved during the resolution of the
+     * first operation on that key. Hence the update chain could contain the prepared updates by
```
Comment (luke-pearson) [inline src/txn/txn.c:1224]: So basically the update chain has transitioned to the committed state, then another transaction has written to it concurrently which is why we see updates from a different txn on the chain?

**Review by luke-pearson — APPROVED** [review]
LGTM.

File: src/txn/txn.c (hunk 1 above)
Comment (bvpvamsikrishna) [reply src/txn/txn.c:1224]: Yes.

**Pull request merged**